app/services/katello/pulp3/repository/file.rb

```diff
@@ -91,6 +91,13 @@ def get_distribution(href)
           raise e if e.code != 404
           nil
         end
+
+        def version_details
+          pulp3_api.repositories_versions_read(repo.version_href)
+        rescue PulpFileClient::ApiError => e
+          raise e if e.code != 404
+          nil
```

Comment:
curious why this was needed?

Reply:
@jlsherrill I saw the same thing used for a similar read operation in the original create PR (for distributions_file_file_read ?). Why check for and trap exceptions for specific pulp operations?

Reply:
it makes sense for deletes IMO, but that is a good question, why it was  being done for reads originally.  I guess looking over that specific case, it was handling the situation where a distribution had been deleted in pulp, but we were still referencing it.  

In these cases, I don't think it can occur until we start deleting repository versions (if we ever do)